mypy/build.py

```diff
@@ -265,7 +265,12 @@ def normpath(path: str, options: Options) -> str:
     # mode would be that a moved file may change its full module
     # name without changing its size, mtime or hash.)
     if options.bazel:
-        return os.path.relpath(path)
+        try:
+            return os.path.relpath(path)
+
+        except ValueError:
+            return os.path.abspath(path)
+
     else:
```

Comment:
This is probably the most critical workaround, since I think this would not work with bazel (when I read the comment above correctly). However, is bazel not Linux only? Would we simple skip the corresponding testcase on Windows systems?

Reply:
Bazel supports Windows these days, though I'd expect that nobody has tried running mypy under Bazel on Windows. Maybe it's fine to require that everything is on a single drive when using Bazel on Windows. Which test case is failing without this change? It's probably fine to skip that test case on Windows.

Reply:
The failing test is `testBazelFlagIgnoresFileChanges `.
This is the error:
```
____________________________________________________________________________________________________________________________ testBazelFlagIgnoresFileChanges ____________________________________________________________________________________________________________________________
[gw5] win32 -- Python 3.7.4 d:\projects\python\stars\dependencies\mypy\mypy-python\scripts\python.exe
data: d:\Projects\Python\stars\dependencies\mypy\main\test-data\unit\check-incremental.test:4121:
d:\Projects\Python\stars\dependencies\mypy\main\mypy\test\testcheck.py:126: in run_case
    self.run_case_once(testcase, ops, step)
d:\Projects\Python\stars\dependencies\mypy\main\mypy\test\testcheck.py:227: in run_case_once
    self.verify_cache(module_data, res.errors, res.manager, res.graph)
d:\Projects\Python\stars\dependencies\mypy\main\mypy\test\testcheck.py:266: in verify_cache
    missing_paths = self.find_missing_cache_files(modules, manager)
d:\Projects\Python\stars\dependencies\mypy\main\mypy\test\testcheck.py:297: in find_missing_cache_files
    if not build.validate_meta(meta, id, path, ignore_errors, manager):
d:\Projects\Python\stars\dependencies\mypy\main\mypy\build.py:1254: in validate_meta
    path = normpath(path, manager.options)
d:\Projects\Python\stars\dependencies\mypy\main\mypy\build.py:269: in normpath
    return os.path.relpath(path)
D:\Programming\Python\Python_3.7.4\lib\ntpath.py:562: in relpath
    path_drive, start_drive))
E   ValueError: path is on mount 'd:', start on mount 'C:'
================================================================================================================ 1 failed, 7910 passed, 236 skipped in 137.68s (0:02:17) ================================================================================================================
```

What would be a better solution: Fallback to absolute path or let the Exception happen and skip the test?

Reply:
Let's skip the test on Windows (but please add comment describing why).